### DISCUSSION TITLE
supabase.auth.getUser() undefined

### DISCUSSION THREAD

**Discussion by BryanGabriel-Atangana** [supabase/supabase#21953] [Questions]
Hello,
I'm building an app with supabase and next js.
I encountered a bug saying getUser() is undefined. I checked the env variables and supabase initialisation and they seemed ok.
![image](https://github.com/supabase/supabase/assets/118725036/9f1addcd-b7b7-41e0-afe6-ca3a594d2aab)

**Comment by akabutnicer** [comment]
The correct function would be `await supabase.auth.getSession()`.

**Comment by ovesco** [comment]
Encountered the same problem. Supabase-js v2.43.1. the `getUser()` method is undefined but the `user()` one is, which doesnt match the typescript types
